packages/i18n-jsx/src/I18n/I18n.tsx

```diff
@@ -1,17 +1,26 @@
 import * as React from 'react'
 import useI18n from '../useI18n'
+import { FArgs, ArgsObj, ArgsObjJsx } from '../utils/types'
 
 interface Props {
   k: string | number
-  args?: (string | number | React.ReactNode)[]
+  args?: FArgs | ArgsObj | ArgsObjJsx
 }
 
 const I18n: React.FC<Props> = props => {
   const { children, k, args } = props
+  const childrenArray = React.Children.toArray(children)
   const i18n = useI18n()
-  const argsParsed = args ? args! : ([] as (string | number | React.ReactNode)[])
 
-  return <>{i18n(k, children as string, ...argsParsed)}</>
+  if (childrenArray.length === 1 && (typeof childrenArray[0] === 'string' || typeof childrenArray[0] === 'number')) {
+    const argsParsed = args ? (Array.isArray(args) ? args : [args]) : ([] as FArgs)
+
+    return <>{i18n(k, children as string, ...(argsParsed as FArgs))}</>
```

Comment:
At first glance it looks like it's calling itself. Maybe just use `useI18n()` instead of assigning to `i18n` vaiarble.